src/ActionModal/components/ActionModal.vue

```diff
@@ -492,20 +493,14 @@ export default {
         account: this.overview.accountInformation
       }
     },
-    // TODO: delete in favor of the multi field in network
-    isMultiDenomNetwork() {
-      return this.balances.length > 1 &&
-        this.balances[0].denom !== this.balances[1].denom
-        ? true
-        : false
-    },
     getDenom() {
       return this.selectedDenom || this.network.stakingDenom
     },
     selectedBalance() {
       const defaultBalance = {
         amount: 0,
-        gasPrice: 4e-7 // TODO: temporary fix for claim rewards for NGM
+        // awful network-specific logic. But how to do it otherwise?
+        gasPrice: gasPricesDictionary(this.getDenom)
```

Comment:
don'T we get this from the API for all the denoms?

Reply:
The problem is the claim's reward case, where there is no balance selected (soon to change with transaction abstraction).

So [this ](https://github.com/luniehq/lunie/blob/7e078d4e345dd2162327e7e9da584e28977a806f/src/ActionModal/components/ActionModal.vue#L503)is the problematic line:

Reply:
if there is no balance selected you can assume the stakingDenom right?

Reply:
That is what I am doing. This is why there are only staking denoms in the dictionary

Reply:
but you are replicating the data in the FE code. you can just use the same data you added to the API here. just default `selectedDenom` to `stakingDenom`. this also reduces the logic here.

Reply:
this makes sense actually. Let me try that

Reply:
I did try it in #3592. There is something a bit weird with one test though.